pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/patch/ObjectDeleteTest.java

```diff
@@ -0,0 +1,183 @@
+
+            int tissueId = Integer.parseInt((String) response.get("tissueId"));
+            Optional<Tissue> maybeCreatedTissue = new TissueDao().get((long) tissueId);
+            Assert.assertTrue(maybeCreatedTissue.isPresent());
+            Tissue createdTissue = maybeCreatedTissue.get();
+            oncHistoryTestUtil.deleteOncHistory(guid2, participantId, instanceName, userEmail, oncHistoryDetailId);
+            oncHistoryTestUtil.assertTissueIsDeleted(guid2, createdTissue, tissueId, ddpInstanceDto, true);
+            oncHistoryTestUtil.assertOncHistoryIsDeleted(guid2, oncHistoryDetail, oncHistoryDetailId, ddpInstanceDto, false, true);
+
+        } catch (Exception e) {
+            throw new RuntimeException("Unexpected exception ", e);
+        }
+
+    }
+
+    @Test
+    public void deleteOncHistoryWithTissueAndSmId() {
+        try {
+            ParticipantDto participantDto = oncHistoryTestUtil.createParticipant(guid3, ddpInstanceDto);
+            guid3 = participantDto.getDdpParticipantIdOrThrow();
+            int participantId = participantDto.getParticipantIdOrThrow();
+            Map<String, Object> response =
+                    (Map<String, Object>) oncHistoryTestUtil.createOncHistory(guid3, participantId, instanceName, userEmail);
+            int oncHistoryDetailId = Integer.parseInt((String) response.get("oncHistoryDetailId"));
+            OncHistoryDetail oncHistoryDetail =
+                    OncHistoryDetail.getOncHistoryDetail(oncHistoryDetailId, ddpInstanceDto.getInstanceName());
+            Assert.assertNotNull(oncHistoryDetail);
+            response = (Map<String, Object>) oncHistoryTestUtil.createTissue(guid3, oncHistoryDetailId, instanceName, userEmail);
+            int tissueId = Integer.parseInt((String) response.get("tissueId"));
+
+            response = (Map<String, Object>) oncHistoryTestUtil.createSmId(guid3, tissueId, instanceName, userEmail, "value-test1");
+            int smIdPk = (int) response.get("smIdPk");
+            SmId smId = new TissueSMIDDao().getBySmIdPk(smIdPk);
+            Optional<Tissue> maybeCreatedTissue = new TissueDao().get((long) tissueId);
+            Assert.assertTrue(maybeCreatedTissue.isPresent());
+            Tissue createdTissue = maybeCreatedTissue.get();
+
+            oncHistoryTestUtil.deleteOncHistory(guid3, participantId, instanceName, userEmail, oncHistoryDetailId);
+            oncHistoryTestUtil.assertSmIdIsDeleted(guid3, tissueId, smIdPk, ddpInstanceDto, smId);
+            oncHistoryTestUtil.assertTissueIsDeleted(guid3, createdTissue, tissueId, ddpInstanceDto, false);
+            oncHistoryTestUtil.assertOncHistoryIsDeleted(guid3, oncHistoryDetail, oncHistoryDetailId, ddpInstanceDto, false, false);
+
+        } catch (Exception e) {
+            throw new RuntimeException("Unexpected exception ", e);
+        }
+
+    }
+
+    @Test
+    public void deleteJustTheSmId() {
```

Comment:
This test creates 2 sm ids for one single onc history and only deletes one of them, then make sure nothing else was deleted